dataStructures/queues/queue.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+)
+
+// Queue data structure, it will be a list of Orders
+type Queue []*Order
+
+// Order data type
+type Order struct {
+	priority     int
+	quantity     int
+	product      string
+	customerName string
+}
+
+// New method creates a new order with required properties
+func (order *Order) New(priority, quantity int, product, customerName string) {
+	order.priority = priority
+	order.quantity = quantity
+	order.product = product
+	order.customerName = customerName
+}
+
+// Add method takes an order and adds it to the queue based on priority
+func (queue *Queue) Add(order *Order) {
+	if len(*queue) == 0 {
+		*queue = append(*queue, order)
+	} else {
+		appended := false
```

Comment:
You can also use `var appended bool` and it uses the "zero value" of bool, which is false.

Reply:
Yeah, that's very true. Thanks!